### ISSUE TITLE
failed processing very short fastq record (core dumped) - v4.3.4

### ISSUE THREAD

**Issue by taniagmangolini** [biocore/sortmerna#321]
Hello, I am facing the following error to run a paired sample (~40 MB each file) in SortmeRNA v4.3.4: **line 102: 111 Aborted (core dumped)**.
The output error log shows the lines below. Anybody have any ideia about what can be happening?
```
*** Error in `sortmerna': free(): invalid next size (fast): 0x0000000000f81e50 ***
======= Backtrace: =========
[0x9659c1]
[0x96df56]
...
[0x938375]
[0x475619]
======= Memory map: ========
00400000-00b28000 r-xp 00000000 00:26 38530                              /opt/conda/bin/sortmerna
00d27000-00d35000 rw-p 00727000 00:26 38530                              /opt/conda/bin/sortmerna
00d35000-00d5f000 rw-p 00000000 00:00 0 
00eae000-00fb3000 rw-p 00000000 00:00 0                                  [heap]
7f90c0000000-7f90c0039000 rw-p 00000000 00:00 0 
7f90c0039000-7f90c4000000 ---p 00000000 00:00 0 
7f90c8000000-7f90c8021000 rw-p 00000000 00:00 0 
...
7ffe429d8000-7ffe429f9000 rw-p 00000000 00:00 0                          [stack]
7ffe429fa000-7ffe429fd000 r--p 00000000 00:00 0                          [vvar]
7ffe429fd000-7ffe429ff000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
```

**Comment by biocodz** [comment]
could you, please, provide the [execution trace](https://github.com/biocore/sortmerna#execution-trace)?

**Comment by taniagmangolini** [comment]
I have run again to get the complete stacktrace and this is the log of execution:

timestamp,message
1648475415154,Found flag: --ref
1648475415154,Found value: silva-bac-16s-id90.fasta of previous flag: --ref
1648475415154,Found flag: --ref
1648475415154,Found value: silva-bac-23s-id98.fasta of previous flag: --ref
1648475415154,Found flag: --ref
1648475415154,Found value: silva-arc-16s-id95.fasta of previous flag: --ref
1648475415154,Found flag: --ref
1648475415154,Found value: silva-arc-23s-id98.fasta of previous flag: --ref
1648475415154,Found flag: --ref
1648475415154,Found value: silva-euk-18s-id95.fasta of previous flag: --ref
1648475415154,Found flag: --ref
1648475415154,Found value: silva-euk-28s-id98.fasta of previous flag: --ref
1648475415154,Found flag: --ref
1648475415154,Found value: rfam-5s-database-id98.fasta of previous flag: --ref
1648475415154,Found flag: --ref
1648475415154,Found value: rfam-5.8s-database-id98.fasta of previous flag: --ref
1648475415154,Found flag: -a
1648475415154,Found value: 30 of previous flag: -a
1648475415154,Found flag: -reads
1648475415154,Found value: /tmp/scratch/bucket/samples/testes/FILE_R1_unmapped_human.fastq of previous flag: -reads
1648475415154,Found flag: -reads
1648475415154,Found value: /tmp/scratch/bucket/samples/testes/FILE_R2_unmapped_human.fastq of previous flag: -reads
1648475415154,Found flag: --aligned
1648475415154,Found value: FILE_rrna of previous flag: --aligned
1648475415154,Found flag: --other
1648475415154,Found value: FILE_nonrrna of previous flag: --other
1648475415154,Found flag: --paired_in
1648475415154,Found value: /tmp/scratch/bucket/samples/testes/FILE_R2_unmapped_human.fastq of previous flag: --paired_in
1648475415154,Found flag: -out2
1648475415154,Found value: /tmp/scratch/bucket/samples/testes/FILE_R2_unmapped_human.fastq of previous flag: -out2
1648475415154,Found flag: --fastx
1648475415154,[process:1460] Processing option: a with value: 30
1648475415154,[opt_a:840] [0;33mWARNING[0m: Option '-a' was Deprecated in favour of '-threads'. Redirecting to 'threads' ...
1648475415154,[process:1460] Processing option: aligned with value: FILE_rrna
1648475415154,[process:1460] Processing option: fastx with value: 
1648475415154,[process:1460] Processing option: other with value: FILE_nonrrna
1648475415154,[process:1460] Processing option: out2 with value: /tmp/scratch/bucket/samples/testes/FILE_R2_unmapped_human.fastq
1648475415154,[process:1460] Processing option: paired_in with value: /tmp/scratch/bucket/samples/testes/FILE_R2_unmapped_human.fastq
1648475415154,[process:1460] Processing option: reads with value: /tmp/scratch/bucket/samples/testes/FILE_R1_unmapped_human.fastq
1648475415154,[opt_reads:97] Processing reads file [1] out of total [2] files
1648475415154,[process:1460] Processing option: reads with value: /tmp/scratch/bucket/samples/testes/FILE_R2_unmapped_human.fastq
1648475415154,[opt_reads:97] Processing reads file [2] out of total [2] files
1648475415154,[process:1460] Processing option: ref with value: silva-bac-16s-id90.fasta
1648475415154,[opt_ref:157] Processing reference [1] out of total [8] references
1648475415154,"[opt_ref:205] File ""/tmp/scratch/silva-bac-16s-id90.fasta"" exists and is readable"
1648475415154,[process:1460] Processing option: ref with value: silva-bac-23s-id98.fasta
1648475415154,[opt_ref:157] Processing reference [2] out of total [8] references
1648475415154,"[opt_ref:205] File ""/tmp/scratch/silva-bac-23s-id98.fasta"" exists and is readable"
1648475415154,[process:1460] Processing option: ref with value: silva-arc-16s-id95.fasta
1648475415154,[opt_ref:157] Processing reference [3] out of total [8] references
1648475415154,"[opt_ref:205] File ""/tmp/scratch/silva-arc-16s-id95.fasta"" exists and is readable"
1648475415154,[process:1460] Processing option: ref with value: silva-arc-23s-id98.fasta
1648475415154,[opt_ref:157] Processing reference [4] out of total [8] references
1648475415154,"[opt_ref:205] File ""/tmp/scratch/silva-arc-23s-id98.fasta"" exists and is readable"
1648475415154,[process:1460] Processing option: ref with value: silva-euk-18s-id95.fasta
1648475415154,[opt_ref:157] Processing reference [5] out of total [8] references
1648475415154,"[opt_ref:205] File ""/tmp/scratch/silva-euk-18s-id95.fasta"" exists and is readable"
1648475415154,[process:1460] Processing option: ref with value: silva-euk-28s-id98.fasta
1648475415154,[opt_ref:157] Processing reference [6] out of total [8] references
1648475415154,"[opt_ref:205] File ""/tmp/scratch/silva-euk-28s-id98.fasta"" exists and is readable"
1648475415154,[process:1460] Processing option: ref with value: rfam-5s-database-id98.fasta
1648475415154,[opt_ref:157] Processing reference [7] out of total [8] references
1648475415154,"[opt_ref:205] File ""/tmp/scratch/rfam-5s-database-id98.fasta"" exists and is readable"
1648475415154,[process:1460] Processing option: ref with value: rfam-5.8s-database-id98.fasta
1648475415154,[opt_ref:157] Processing reference [8] out of total [8] references
1648475415154,"[opt_ref:205] File ""/tmp/scratch/rfam-5.8s-database-id98.fasta"" exists and is readable"
1648475415154,[process:1480] === Options processing done ===
1648475415154,[process:1481] Alignment type: [best:1 num_alignments:1 min_lis:2 seeds:2]
1648475415154,"[validate_kvdbdir:1221] 'workdir' option was not provided. Using USERDIR to set the working directory: """""
1648475415154,"[validate_kvdbdir:1227] Key-value DB location ""/home/biodocker/sortmerna/run/kvdb"""
1648475415154,"[validate_kvdbdir:1263] Creating KVDB directory: ""/home/biodocker/sortmerna/run/kvdb"""
1648475415154,"[validate_idxdir:1193] Using index directory: ""/home/biodocker/sortmerna/run/idx"""
1648475415154,[validate_idxdir:1201] Created index directory - OK
1648475415154,"[validate_readb_dir:1285] Using split reads directory : ""/home/biodocker/sortmerna/run/readb"""
1648475415154,[validate_readb_dir:1293] Created split reads directory - OK
1648475415154,[main:62] Running command:
1648475595824,sortmerna --ref silva-bac-16s-id90.fasta --ref silva-bac-23s-id98.fasta --ref silva-arc-16s-id95.fasta --ref silva-arc-23s-id98.fasta --ref silva-euk-18s-id95.fasta --ref silva-euk-28s-id98.fasta --ref rfam-5s-database-id98.fasta --ref rfam-5.8s-database-id98.fasta -a 30 -reads /tmp/scratch/bucket/samples/testes/FILE_R1_unmapped_human.fastq -reads /tmp/scratch/bucket/samples/testes/FILE_R2_unmapped_human.fastq --aligned FILE_rrna --other FILE_nonrrna --paired_in /tmp/scratch/bucket/samples/testes/FILE_R2_unmapped_human.fastq -out2 /tmp/scratch/bucket/samples/testes/FILE_R2_unmapped_human.fastq --fastx 
1648475595824,[build_index:1127] ==== Index building started ====
1648475595824,[build_index:1190] Begin indexing file silva-bac-16s-id90.fasta of size: 19437013 under index name /home/biodocker/sortmerna/run/idx/15734375058464002811
1648475595824,  done.
1648475595824,[build_index:1190] Begin indexing file silva-bac-23s-id98.fasta of size: 12911743 under index name /home/biodocker/sortmerna/run/idx/17299952793705614139
1648475595824,  done.
1648475595824,[build_index:1190] Begin indexing file silva-arc-16s-id95.fasta of size: 3893959 under index name /home/biodocker/sortmerna/run/idx/3436099190853847617
1648475595824,  done.
1648475595824,[build_index:1190] Begin indexing file silva-arc-23s-id98.fasta of size: 752022 under index name /home/biodocker/sortmerna/run/idx/3400685301612210653
1648475595824,  done.
1648475595824,[build_index:1190] Begin indexing file silva-euk-18s-id95.fasta of size: 13259584 under index name /home/biodocker/sortmerna/run/idx/2700646386527218729
1648475595824,  done.
1648475595824,[build_index:1190] Begin indexing file silva-euk-28s-id98.fasta of size: 14945070 under index name /home/biodocker/sortmerna/run/idx/1845323523482939374
1648475595824,  done.
1648475595824,[build_index:1190] Begin indexing file rfam-5s-database-id98.fasta of size: 8525326 under index name /home/biodocker/sortmerna/run/idx/13019673092862722585
1648475595824,  done.
1648475595824,[build_index:1190] Begin indexing file rfam-5.8s-database-id98.fasta of size: 2280449 under index name /home/biodocker/sortmerna/run/idx/2169995244134016533
1648475595824,  done.
1648475595824,[build_index:2108] ==== Done index building in 180.624 sec ====
1648475595824,[init:108] Readfeed init started
1648475595824,"[define_format:915] file: ""/tmp/scratch/bucket/samples/testes/FILE_R1_unmapped_human.fastq"" is FASTQ flat ASCII"
1648475595832,/tmp/tmp.zw8QiuqMQ/batch-file-temp: line 101:    94 Aborted                 (core dumped) sortmerna --ref silva-bac-16s-id90.fasta --ref silva-bac-23s-id98.fasta --ref silva-arc-16s-id95.fasta --ref silva-arc-23s-id98.fasta --ref silva-euk-18s-id95.fasta --ref silva-euk-28s-id98.fasta --ref rfam-5s-database-id98.fasta --ref rfam-5.8s-database-id98.fasta -a 30 -reads /tmp/scratch/bucket/samples/testes/FILE_R1_unmapped_human.fastq -reads /tmp/scratch/bucket/samples/testes/FILE_R2_unmapped_human.fastq --aligned FILE_rrna --other FILE_nonrrna --paired_in /tmp/scratch/bucket/samples/testes/FILE_R2_unmapped_human.fastq -out2 /tmp/scratch/bucket/samples/testes/FILE_R2_unmapped_human.fastq --fastx 2> errors.txt

**Comment by biocodz** [comment]
The program has a problem reading your reads files, but I cannot say the reason exactly from the trace.
What's the output of
```
ls -l /home/biodocker/sortmerna/run/readb
```
?

**Comment by taniagmangolini** [comment]
I have checked here there aren't any files in /home/biodocker/sortmerna/run/readb

**Comment by taniagmangolini** [comment]
I was examining the code to underestand what is going on and seems that the problem is in readfeed.cpp because the next function count_reads() is not being called. 
Seems that sortmerna is not being able to read the files R1 and R2. So I have applied a downsampling to the reads until I get where the problem happens and I could find that that problem was the first read (I don't have idea why). But I just removed the first read from R1 and R2 files and the problem was solved.
This is the read I have removed:
- In R1 file:
@NB561001:418:HGLM7BFX4:1:11101:1060:10057/1
GACTATATGGAT
+
EEEEAEEEEEEA

- In R2 file:
- @NB561001:418:HGLM7BFX4:1:11101:1060:10057/2
AGGGGTTGGCTTGAAACCAGCTTTGGGGGGTTCGATTCCTTCCTTTTTTGTCTAGATTTTATGTATACGGGTTCTTCGAATGTGTGGTAGGGTGGGGGGCATCCATATAGTCACTCCAGGTCTTTCTCTTATACA
+
EAEEEEEEEEEEEEEAEEEEE/EEEEE/EEEAEEEAEE/EEAEEEEAA<<AE<EEEEAEAEAAEA/EEEEE/6AEEEEE<EEA<E/AEEEEEEEE<AEEE/EAAE/EA<EAAA<<</AE<AEE/</<<<AA</<<

**Comment by biocodz** [comment]
Thanks a lot for the elucidation of this problem! 
In the coming release we'll add a better validation for such cases.

**Comment by biocodz** [comment]
related to [issue 290](https://github.com/biocore/sortmerna/issues/290). Working on the fix...

**Comment by biocodz** [comment]
pre-release [4.3.5-pre](https://github.com/biocore/sortmerna/releases/tag/v4.3.5-pre) includes the fix

**Comment by biocodz** [comment]
fixed in [v4.3.6](https://github.com/biocore/sortmerna/releases/tag/v4.3.6)
